gui/msgwin.c

```diff
@@ -0,0 +1,233 @@
+/**
+ * @file
+ * Message Window
+ *
+ * @authors
+ * Copyright (C) 2021 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @page gui_msgwin Message Window
+ *
+ * Message Window for displaying messages and text entry.
+ */
+
+#include "config.h"
+#include <stddef.h>
+#include "mutt/lib.h"
+#include "core/lib.h"
+#include "color.h"
+#include "curs_lib.h"
+#include "mutt_curses.h"
+#include "mutt_window.h"
+
+struct MuttWindow *MessageWindow = NULL; ///< Message Window, ":set", etc
+
+/**
+ * struct MsgWinPrivateData - Private data for the Message Window
+ */
+struct MsgWinPrivateData
+{
+  enum ColorId color; ///< Colour for the text, e.g. #MT_COLOR_MESSAGE
+  char *text;         ///< Cached display string
+};
+
+/**
+ * msgwin_recalc - Recalculate the display of the Message Window
+ * @param win Message Window
+ */
+static int msgwin_recalc(struct MuttWindow *win)
+{
+  if (!window_is_focused(win))
+    return 0;
+
+  // If nobody is using it, we'll take charge
+  win->actions |= WA_REPAINT;
+  mutt_debug(LL_DEBUG5, "recalc done, request WA_REPAINT\n");
+  return 0;
+}
+
+/**
+ * msgwin_repaint - Redraw the Message Window
+ * @param win Message Window
+ */
+static int msgwin_repaint(struct MuttWindow *win)
+{
+  if (!mutt_window_is_visible(win) || !window_is_focused(win))
```

Comment:
Realized the code I gave you was logically incorrect. Conditional should be:

```C
if (!mutt_window_is_visible(win) || window_is_focused(win))
  return 0;
```

Since we want to make sure the window **isn't** focused.

Reply:
urgh!  yes :-)
fixed